src/std/srfi/srfi-19.scm

```diff
@@ -253,7 +258,25 @@
 ;;; the time structure; creates the accessors, too.
 ;;; wf: changed to match srfi documentation. uses mzscheme structures & inspectors
 ;;; vyzo: change to use gerbil defstruct
-(defstruct time (type nanosecond second))
+;;; fare: added checking in make-time
+(defstruct time (type nanosecond second)
+  constructor: init!)
+
+(defmethod {init! time}
+  (lambda (time type second nanosecond)
+    (case type
+      ((time-tai time-utc time-monotonic time-thread time-process time-duration #f)
+       (set-time-type! time type))
```

Comment:
be consistent pls, `(set! (time-type time) type)`

Reply:
Yeah, oops. But since the rest of the file uses the opposite convention, I'll use the opposite convention in the rest of this function instead. (I think I started using `set!` but used the other after I got an error, probably due to a typo.)

Reply:
either way works, as long as we are consistent :)

Reply:
Done.